clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaApplicationClientImpl.java

```diff
@@ -41,6 +42,7 @@
 
     private final Vertx vertx;
     private final KafkaConsumerConfigProperties consumerConfig;
+    private Supplier<KafkaConsumer<String, Buffer>> kafkaConsumerSupplier;
```

Comment:
IMHO this should be a function accepting a String (type) so that you can use a default implementation here that invokes KafkaConsumer.create(vertx, String) ...

Reply:
For the test we need the method with parameters as below to create a Kafka consumer
`KafkaConsumer.create(vertx, consumer)`

And in the client implementation the below one. 
`KafkaConsumer.create(vertx,  configMap)`

Both are methods with different signature. Sorry I didn't get it what you mean by to use a default implementation that invokes `KafkaConsumer.create(vertx, String)`